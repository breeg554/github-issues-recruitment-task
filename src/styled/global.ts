import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,button,input,textarea{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body{
    background-color: #E5E5E5;
  }
`;
