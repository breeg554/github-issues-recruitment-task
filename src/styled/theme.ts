export type ThemeType = typeof theme; // This is the type definition for my theme object.

export const theme = {
  colors: {
    core: {
      primary: "#24292F",
      secondary: "#166CD7",
      tertiary: "#6F7781",
    },
    hover: {
      primary: "#353a3F",
      secondary: "#288ef9",
      tertiary: "#9F9992",
    },
  },
  mediaQ: {
    sm: "@media(min-width: 576px)",
    md: "@media(min-width: 768px)",
    lg: "@media(min-width: 992px)",
    xl: "@media(min-width: 1200px)",
  },
};
