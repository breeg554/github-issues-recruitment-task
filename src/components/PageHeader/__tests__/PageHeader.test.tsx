import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { PageHeader } from "..";
import SearchProvider from "../../../context/Search";
import { theme } from "../../../styled/theme";

describe("Page header component", () => {
  it("renders out and match the snapshot", () => {
    const { container } = render(
      <Router>
        <ThemeProvider theme={theme}>
          <SearchProvider>
            <PageHeader />
          </SearchProvider>
        </ThemeProvider>
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
