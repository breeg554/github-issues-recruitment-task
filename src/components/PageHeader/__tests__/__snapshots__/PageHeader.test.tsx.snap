// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page header component renders out and match the snapshot 1`] = `
<div>
  <header
    class="sc-iCfMLu hQHjyI"
  >
    <div>
      <a
        aria-label="Home page"
        class="github-logo"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="octicon octicon-mark-github"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
      <div
        class="sc-gsDKAQ fIhKta"
      >
        <input
          aria-label="Search users and repositories"
          class="sc-dkPtRN bRWSnv search-input"
          placeholder="Search"
          value=""
        />
        <button
          class="reset-search"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-x"
            fill="currentColor"
            height="12"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="12"
          >
            <path
              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</div>
`;
